packages/datatrak-web/src/layout/Header/HeaderLeft.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React, { useState } from 'react';
+import { useMatch } from 'react-router';
+import { ROUTES } from '../../constants';
+import { CancelConfirmModal } from '../../components';
+import { DesktopHeaderLeft } from './DesktopHeaderLeft';
+import { MobileHeaderLeft } from './MobileHeaderLeft';
+
+export const HeaderLeft = () => {
+  const [surveyCancelModalIsOpen, setIsOpen] = useState(false);
+  const isSurveyScreen = !!useMatch(ROUTES.SURVEY_SCREEN);
+  const isSuccessScreen = !!useMatch(ROUTES.SURVEY_SUCCESS);
+
+  const onClickLogo = e => {
+    if (isSurveyScreen && !isSuccessScreen) {
+      e.preventDefault();
+      setIsOpen(true);
+    }
+  };
+
+  const onClose = () => {
+    setIsOpen(false);
+  };
+  return (
+    <>
+      <DesktopHeaderLeft onClickLogo={onClickLogo} />
+      <MobileHeaderLeft onClickLogo={onClickLogo} />
```

Comment:
Looks like both of these `xxxHeaderLeft` elements check for `${({ theme }) => theme.breakpoints.up('md')}`

How about doing it just once here with a `useMediaQuery` to avoid `render`ing both elements?

```tsx
const isMobile = useMediaQuery(...)
const HeaderComponent = isMobile ? MobileHeaderLeft : DesktopHeaderLeft;

return (
  <>
    <HeaderComponent onClickLogo={onClickLogo} />
    <CancelConfirmModal />
```

Ditto for `ActivityFeedSection.tsx` and `TasksSection.tsx`